lib/puppet/provider/vcsrepo/git.rb

```diff
@@ -111,9 +111,16 @@ def default_url
     end
   end
 
+  def correct_default_url?
+      File.readlines(File.join(@resource.value(:path), '.git', 'config')).grep(/#{Regexp.escape(default_url)}/).any?
+  end
+
   def working_copy_exists?
     if @resource.value(:source) and File.exists?(File.join(@resource.value(:path), '.git', 'config'))
-      File.readlines(File.join(@resource.value(:path), '.git', 'config')).grep(/#{Regexp.escape(default_url)}/).any?
+        if !correct_default_url?
+            update_remotes
```

Comment:
We should avoid making modifications to the repository in the working_copy_exists? function.